bindings/java/src/main/java/org/ucx/jucx/ucp/UcpContext.java

```diff
@@ -41,8 +44,37 @@ public void close() {
         this.setNativeId(null);
     }
 
+    /**
+     * Allocates a memory segment with the network resources associated
+     * with it. The network stack associated with an application context
+     * can typically send and receive data from the mapped memory without
+     * CPU intervention; some devices and associated network stacks
+     * require the memory to be mapped to send and receive data.
+     *
+     * @param size - since java uses int for array indexes,
+     *               maximum memory allocation size is limited to 2Gb
+     * @return
+     */
+    public UcpMemory allocateMemory(int size) {
```

Comment:
why do we need to have memory allocation API ? isnt it enough to have registration?